skiko/src/awtTest/kotlin/org/jetbrains/skiko/SkiaSwingLayerTest.kt

```diff
@@ -0,0 +1,87 @@
+package org.jetbrains.skiko
+
+import kotlinx.coroutines.delay
+import org.jetbrains.skia.Canvas
+import org.jetbrains.skia.Paint
+import org.jetbrains.skia.Rect
+import org.jetbrains.skiko.swing.SkiaSwingLayer
+import org.jetbrains.skiko.util.ScreenshotTestRule
+import org.jetbrains.skiko.util.uiTest
+import org.junit.Rule
+import java.awt.Color
+import java.awt.Dimension
+import java.awt.Graphics
+import java.awt.Graphics2D
+import java.awt.RenderingHints
+import javax.swing.JComponent
+import javax.swing.JFrame
+import javax.swing.JLayeredPane
+import javax.swing.WindowConstants
+import kotlin.test.Test
+
+@OptIn(ExperimentalSkikoApi::class)
+class SkiaSwingLayerTest {
+
+    @get:Rule
+    val screenshots = ScreenshotTestRule()
+
+    @Test
+    fun `overlapped popup`() = uiTest {
+        val skikoView = fillSkikoView()
+        val skiaLayer = SkiaSwingLayer(skikoView, SkiaLayerProperties().copy(renderApi = renderApi)).apply {
+            setBounds(0, 0, 200, 200)
+        }
+
+        val window = object : JFrame() {
+            override fun dispose() {
+                skiaLayer.dispose()
+                super.dispose()
+            }
+        }
+        try {
+            val popup = object : JComponent() {
+                init {
+                    isOpaque = false
+                    setBounds(50, 50, 50, 50)
+                }
+
+                override fun paintComponent(g: Graphics?) {
+                    val scratchGraphics = g?.create() as? Graphics2D ?: return
+                    try {
+                        scratchGraphics.setRenderingHint(
+                            RenderingHints.KEY_ANTIALIASING,
+                            RenderingHints.VALUE_ANTIALIAS_ON
```

Comment:
Better to disable antialising, it can behave differently on different systems and the screenshots will be different.

It is the case with Skia, not sure about Graphics2D. It still succeeds on Windows.